### PR TITLE
En_Wonder_Item

### PR DESCRIPTION
**Pull request by petrie911**
Wonderful, isn't it? Also a bit of cleanup and docs on Wonder_Talk2.

### CODE REVIEW

File: src/overlays/actors/ovl_En_Wonder_Item/z_en_wonder_item.c [hunk 1]
```diff
@@ -26,25 +47,316 @@ const ActorInit En_Wonder_Item_InitVars = {
...
+    };
+    s32 pad;
+    s16 colTypeIndex;
+    EnWonderItem* this = THIS;
+    s16 rotZover10;
+    s16 tagIndex;
+
+    osSyncPrintf("\n\n");
+    // Mysterious mystery
+    osSyncPrintf("\x1b[32m☆☆☆☆☆ 不思議不思議まか不思議 \t   ☆☆☆☆☆ %x\n\x1b[m", this->actor.params);
```
Comment (Sirius902) [inline src/overlays/actors/ovl_En_Wonder_Item/z_en_wonder_item.c:112]: Don't forget to use the VT macros! There's a few other places in the file too. :)

**Review by Sirius902 — CHANGES_REQUESTED** [review]
Nice. 👍

File: src/overlays/actors/ovl_En_Wonder_Item/z_en_wonder_item.c (hunk 1 above)
Comment (petrie911) [reply src/overlays/actors/ovl_En_Wonder_Item/z_en_wonder_item.c]: Handled.

**Review by Kenix3 — APPROVED** [review]
Looks good to me!

File: src/overlays/actors/ovl_En_Wonder_Item/z_en_wonder_item.c [hunk 2]
```diff
@@ -26,25 +48,312 @@ const ActorInit En_Wonder_Item_InitVars = {
...
+            sTagPointsOrdered[tagIndex] = this->actor.posRot.pos;
+            Actor_Kill(&this->actor);
+            break;
+        case WONDERITEM_PROXIMITY_SWITCH:
+            this->updateFunc = EnWonderItem_ProximitySwitch;
+            break;
+        case WONDERITEM_BOMB_SOLDIER:
+            Collider_InitCylinder(globalCtx, &this->collider);
+            Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+            this->collider.body.bumper.flags = WONDERITEM_DAMAGE_SLINGSHOT;
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Wonder_Item/z_en_wonder_item.c:182]: wont these defines be provided by the collision check system? dont think we should make actor specific ones

File: src/overlays/actors/ovl_En_Wonder_Item/z_en_wonder_item.c [hunk 3]
```diff
@@ -26,25 +48,312 @@ const ActorInit En_Wonder_Item_InitVars = {
...
+    static s16 debugTextColors[] = {
+        255, 255, 0,   255, 0,   255, 0,   255, 255, 255, 0,   0, 0, 255, 0,   0, 0, 255, 128, 128,
+        128, 128, 128, 0,   128, 0,   128, 0,   128, 0,   128, 0, 0, 0,   128, 0, 0, 0,   128,
+    }; // These seem to be mistyped. Logically they should be s16[13][3] and be indexed as [wonderIndex][i]
+    s32 pad;
+    EnWonderItem* this = THIS;
+    s32 wonderIndex;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wonder_Item/func_80B38AD8.s")
+    DECR(this->timer);
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Wonder_Item/z_en_wonder_item.c:338]: decr

File: src/overlays/actors/ovl_En_Wonder_Item/z_en_wonder_item.c (hunk 2 above)
Comment (petrie911) [reply src/overlays/actors/ovl_En_Wonder_Item/z_en_wonder_item.c:182]: Not for a while. I think we'll need a lot more actors decomped before it becomes clear which dFlag combinations will want macros. In the meantime, I don't see any harm in including these for readability. They can be easily replaced later.

Comment (fig02) [reply src/overlays/actors/ovl_En_Wonder_Item/z_en_wonder_item.c:182]: people often refer to other files to see what conventions we currently have, and may see this and think making their own defines is required. i would prefer a comment that is removed later honestly

**Review by fig02 — APPROVED** [review]

**Review by NEstelami — APPROVED** [review]

File: src/overlays/actors/ovl_En_Wonder_Item/z_en_wonder_item.c [hunk 4]
```diff
@@ -26,25 +40,314 @@ const ActorInit En_Wonder_Item_InitVars = {
...
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wonder_Item/func_80B387F0.s")
+void EnWonderItem_RollDrop(EnWonderItem* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wonder_Item/func_80B38824.s")
+    if ((this->actor.xzDistFromLink < 50.0f) && (player->invincibilityTimer < 0) &&
+        (fabsf(this->actor.posRot.pos.y - player->actor.posRot.pos.y) < 30.0f)) {
+        EnWonderItem_DropCollectible(this, globalCtx, 1);
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Wonder_Item/z_en_wonder_item.c:317]:
```suggestion
        EnWonderItem_DropCollectible(this, globalCtx, true);
```

File: src/overlays/actors/ovl_En_Wonder_Item/z_en_wonder_item.h [hunk 5]
```diff
@@ -6,11 +6,58 @@
...
+    /* 4 */ WONDERITEM_UNUSED,
+    /* 5 */ WONDERITEM_MULTITAG_ORDERED,
+    /* 6 */ WONDERITEM_TAG_POINT_ORDERED,
+    /* 7 */ WONDERITEM_PROXIMITY_SWITCH,
+    /* 8 */ WONDERITEM_BOMB_SOLDIER,
+    /* 9 */ WONDERITEM_ROLL_DROP
+} EnWonderItemModes;
+
+typedef enum {
+    /*  0 */ WONDERITEM_NUTS,
```
[the remark this answers is not in the document]
Comment (petrie911) [reply src/overlays/actors/ovl_En_Wonder_Item/z_en_wonder_item.h]: Now WONDERITEM_DROP_

File: src/overlays/actors/ovl_En_Wonder_Item/z_en_wonder_item.c [hunk 6]
```diff
@@ -26,25 +40,314 @@ const ActorInit En_Wonder_Item_InitVars = {
...
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wonder_Item/EnWonderItem_Update.s")
+    wonderIndex = this->wonderMode;
+    if (this->wonderMode > 12) {
+        wonderIndex = 0;
+    }
+    if (BREG(0) != 0) {
+        DebugDisplay_AddObject(this->actor.posRot.pos.x, this->actor.posRot.pos.y, this->actor.posRot.pos.z,
+                               this->actor.posRot.rot.x, this->actor.posRot.rot.y, this->actor.posRot.rot.z, 1.0f, 1.0f,
+                               1.0f, debugTextColors[wonderIndex], debugTextColors[wonderIndex + 1],
+                               debugTextColors[wonderIndex + 2], 255, 4, globalCtx->state.gfxCtx);
```
[the remark this answers is not in the document]
Comment (petrie911) [reply src/overlays/actors/ovl_En_Wonder_Item/z_en_wonder_item.c]: debugArrowColors it is.

**Review by Roman971 — APPROVED** [review]

**Pull request merged**